lib/stub_ui/lib/src/ui/text.dart

```diff
@@ -1011,6 +1011,185 @@ enum BoxWidthStyle {
   max,
 }
 
+/// Data class that holds the metrics for a single line in the [Paragraph].
+class LineMetrics {
+
+  @pragma('vm:entry-point')
+  LineMetrics._(
```

Comment:
this this constructor have to be private?

Might be worth having this be a public constructor with named arguments and having a private constructor with positional arguments for calling from C++. Having a way to create this class from Dart seems valuable (e.g. for testing).